applications/minotari_console_wallet/src/ui/components/burn_tab.rs

```diff
@@ -497,22 +497,29 @@ impl<B: Backend> Component<B> for BurnTab {
 
         let rx_option = self.burn_result_watch.take();
         if let Some(rx) = rx_option {
-            trace!(target: LOG_TARGET, "{:?}", (*rx.borrow()).clone());
-            let status = match (*rx.borrow()).clone() {
-                UiTransactionBurnStatus::Initiated => "Initiated",
-                UiTransactionBurnStatus::Error(e) => {
-                    self.error_message = Some(format!("Error sending transaction: {}, Press Enter to continue.", e));
-                    return;
-                },
-                UiTransactionBurnStatus::TransactionComplete((
-                    _proof_id,
-                    _reciprocal_claim_public_key,
-                    _serialized_proof,
-                )) => {
-                    self.success_message =
-                        Some("Transaction completed successfully!\nPlease press Enter to continue".to_string());
-                    return;
-                },
+            {
+                // Ensure the watch borrow is dropped immediately after use
+                trace!(target: LOG_TARGET, "{:?}", (*rx.borrow()).clone());
```

Comment:
This is dropping the borrow after this line ends (the watch::Ref isn't put a variable). Also if you look at the macro expansion of trace! debug! etc it already scopes everything.

```rust
{
        let lvl = ::log::Level::Trace;
        if lvl <= ::log::STATIC_MAX_LEVEL && lvl <= ::log::max_level() { // <---- start scope
            ::log::__private_api::log(
                ::log::__private_api::format_args!("{:?}",
                                                   (*rx.borrow()).clone()
                ),
                lvl,
                &(LOG_TARGET, ::log::__private_api::module_path!( ), ::log::__private_api::file!( )),
                ::log::__private_api::line!( ),
                ::log::__private_api::Option::None,
            );
        }
    } // <---- end scope
```

Also shouldn't need to clone the value since we're just logging the value
```suggestion
                trace!(target: LOG_TARGET, "{:?}", rx.borrow()));
```

Reply:
Yeah, I did not look inside the macro...

Reply:
The macro expansion isn't really relevant though, the scoping would not be needed in either case because the borrow (watch::Ref) would drop immediately after `format_args!`